crates/starknet-os/src/starknet/core/os/contract_class/compiled_class_hash_objects.rs

```diff
@@ -71,6 +92,18 @@ impl BytecodeSegmentedNode {
             }
         }
     }
+
+    pub fn hash(&self) -> Result<FieldElement, SyscallExecutionError> {
+        let mut felts = Vec::new();
+
+        for segment in &self.segments {
+            felts.push(FieldElement::from(segment.segment_length.0));
+            felts.push(segment.inner_structure.hash()?);
+        }
+
+        // hash([segment_lent, hash(inner_struct)]) + 1
```

Comment:
Remove commented code. If this is supposed to be an explanation, put it in words + optional pseudo-code.